tests/conftest.py

```diff
@@ -100,3 +101,68 @@ def select_by_name(dash_duo, selector, name):
 def get_options(dash_duo, selector):
     parameter_selector_input = dash_duo.find_element(selector)
     return parameter_selector_input.text.split("\n")
+
+
+def setup_plugin(dash_duo, name, pluginClass, window_size=(1024, 3600)):
+    app = dash.Dash(name)
+    plugin = pluginClass(app, project_identifier=None)
+    layout = plugin.layout
+    app.layout = layout
+    dash_duo.start_server(app)
+    windowsize = window_size
+    dash_duo.driver.set_window_size(*windowsize)
+    return plugin
+
+
+def select_ensemble(dash_duo, plugin, wanted_ensemble_name=None):
+    """tries to select, i.e. click, the ensemble given by ensemble_name, and
+    clicks the first one if no name is given. It returns the name of the
+    selected ensemble."""
+    ensemble_selector_id = plugin.uuid("ensemble-multi-selector")
+    if not wanted_ensemble_name:
+        first_ensemble_name = select_first(dash_duo, "#" + ensemble_selector_id)
+        dash_duo.wait_for_contains_text(
+            "#" + plugin.uuid("selected-ensemble-dropdown"),
+            first_ensemble_name,
+            timeout=4,
+        )
+        return first_ensemble_name
+
+    options_selector_prefix = f"#{ensemble_selector_id}"
+    select_by_name(dash_duo, options_selector_prefix, wanted_ensemble_name)
+    dash_duo.wait_for_contains_text(
+        "#" + plugin.uuid("selected-ensemble-dropdown"),
+        wanted_ensemble_name,
+        timeout=4,
+    )
+    return wanted_ensemble_name
+
+
+def _select_param_or_response(dash_duo, plugin, containerSelector, option_name):
+    select_by_name(dash_duo, containerSelector, option_name)
+    dash_duo.wait_for_element(f"#{plugin.uuid(option_name)}")
```

Comment:
This indirection is not really required, I would remove it and use just `select_by_name`

Reply:
yass, you're right - done!